transport/server.cc

```diff
@@ -618,6 +618,11 @@ future<> cql_server::connection::process_request() {
                     f.length, mem_estimate, _server._max_request_size));
         }
 
+        if (_server._requests_serving > _server._config.max_concurrent_requests) {
```

Comment:
I have second thoughts about it.
I think it's very dangerous to leave it like this, @psarna. There is no such thing as "reasonable value" - there is a robust and not robust code.
This one is not robust.
 
The proper way to do this trick is to define the `max_concurrent_requests` default value to be such that can never be achieved by `_requests_serving`. 

One way to do that is to define `_requests_serving` as u32 and `max_concurrent_requests` be u64 and then initialize `max_concurrent_requests` to limits<u64>::max().

This way this code would always work no matter what.

Reply:
`max_concurrent_requests` does not need to be `u64`. If `_requests_serving` is u32 and `_server._config.max_concurrent_requests` is equal to u32::max, then this condition is always false, so it never sheds any requests. Effectively we're in the same situation even now, because it's not feasible for any modern shard to handle over 4 billion concurrent requests at a time -- but, I already have a small pull request with amendments to this one (https://github.com/scylladb/scylla/pull/7299), so I'll just add a commit which makes `_requests_serving` a u32.

Reply:
According to https://en.cppreference.com/w/cpp/types/numeric_limits the limits<uint32>::max() is achievable by uint32 value so my point holds.

Reply:
Of course it's achievable -- but assuming that `limit` is equal to `u32::max` and `x` is u32, a condition `x > limit` is never true, even for `x == u32::max`. (I hope this message ends up properly formatted, typing from my phone)

Reply:
Oh, you are right! Missed that. My bad.
So, yes, if we make `_requests_serving` u32 we should be totally safe.

Reply:
Hmmm... @psarna Are you planning to send a followup patch that would make _requests_serving u32 to address my comments above?

It would be better if this patch is reverted and you make the change in `_requests_serving` first otherwise it may break the `git bisect` if you try to catch a bug we mentioned above in the future.

Reply:
A followup was already sent yesterday: https://github.com/scylladb/scylla/pull/7299